src/app/business/profile/profile.component.ts

```diff
@@ -68,12 +72,51 @@ export class ProfileComponent implements OnInit {
     }
     getProfiles() {
         this.ProfileService.getProfiles().subscribe((res) => {
-            this.profiles = res.json();
-            this.profiles.forEach(item=>{
+            let str = res.json();
+            this.allProfileNameForCheck = [];
+            str.forEach(item=>{
                 if(item.storageType && item.storageType == "file"){
                     delete item.replicationProperties;
                 }
+                this.allProfileNameForCheck.push(item.name);
             })
+            this.getPools(str);
         });
     }
+    getPools(str) {
+        this.PoolService.getPools().subscribe((res) => {
+            this.pools = res.json();
+            str.forEach(item => {
+                item.totalFreeCapacity = this.getSumCapacity(this.pools, 'free',item);
+                item.totalCapacity = this.getSumCapacity(this.pools, 'total',item);
+            });
+            this.profiles = str;
+        },
+        err=>{
+            this.profiles = str;
+        });
+    }
+    getSumCapacity(pools, FreeOrTotal,data) {
+        let SumCapacity: number = 0;
+        let newPools = lodash.cloneDeep(pools);
+        newPools = newPools.filter(item=>{
+            return item.storageType == data.storageType;
+        })
+        let arrLength = newPools.length;
+        for (let i = 0; i < arrLength; i++) {
+            let valid = data && data["provisioningProperties"].ioConnectivity.accessProtocol && data["provisioningProperties"].ioConnectivity.accessProtocol.toLowerCase() 
+            == newPools[i].extras.ioConnectivity.accessProtocol;
+            let blockValid = data["provisioningProperties"].dataStorage.provisioningPolicy == newPools[i].extras.dataStorage.provisioningPolicy;
```

Comment:
the UI code should NOT be aware of storage types...it should just use the types given by backend

Reply:
This has been added to backlog by updating #84

Reply:
LG...AB